buildvertical.targets

```diff
@@ -5,106 +5,82 @@
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
   </PropertyGroup>
 
- <!-- Required for FindBestConfiguration task -->
- <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
+  <!-- Import configuration data model -->
+  <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
 
- <Target Name="AnnotateProjects"
-         BeforeTargets="BuildAllProjects"
-         Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
-   <!-- Clear ProjectWithConfiguration to prevent circular dependency -->
-   <ItemGroup><ProjectWithConfiguration Remove="@(ProjectWithConfiguration)" /></ItemGroup>
-   <MSBuild Targets="AnnotateProjectsWithConfiguration" 
-            Projects="@(Project)"> 
-     <Output TaskParameter="TargetOutputs" 
-             ItemName="ProjectWithConfiguration" /> 
-   </MSBuild> 
-   <ItemGroup>
-     <Project Remove="@(Project)" />
-     <Project Include="@(ProjectWithConfiguration)" />
-   </ItemGroup>
- </Target>
+  <!-- Runs during traversal to select which projects and configurations of those projects to build -->
+  <Target Name="AnnotateAndFilterProjects"
+          BeforeTargets="BuildAllProjects"
+          Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
 
- <Target Name="AnnotateProjectsWithConfiguration" 
-         Returns="@(ProjectWithConfiguration)"> 
-    <ItemGroup> 
-       <ProjectWithConfiguration Include="$(MSBuildProjectFullPath)"> 
-         <AdditionalProperties Condition="'$(_BestConfiguration)' != ''">Configuration=$(_BestConfiguration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
-         <UndefineProperties Condition="'$(_BestConfiguration)' == ''">Configuration;%(ProjectWithConfiguration.UndefineProperties)</UndefineProperties>
-       </ProjectWithConfiguration>
-       <ProjectWithConfiguration Remove="@(OmitProject)" />
-     </ItemGroup>
-  </Target>
-
-  <!-- The initial shaking target for trimming down applicable projects for specified vertical -->
-  <Target Name="DetermineProjectsConfiguration"
```

Comment:
Yeah, you're right.  I'll fix up that commit message.  I think I got half way through and realized it was no longer needed.